sxg_rs/src/signature/js_signer.rs

```diff
@@ -56,13 +57,13 @@ impl JsSigner {
 
 #[async_trait(?Send)]
 impl Signer for JsSigner {
-    async fn sign(&self, message: &[u8]) -> Result<Vec<u8>,String> {
+    async fn sign(&self, message: &[u8]) -> Result<Vec<u8>> {
         let a = Uint8Array::new_with_length(message.len() as u32);
         a.copy_from(&message);
         let this = JsValue::null();
-        let sig = self.js_function.call1(&this, &a).map_err(|_| "Bad call")?;
+        let sig = self.js_function.call1(&this, &a).map_err(|_| Error::msg("JavaScript signer throws an error."))?;
         let sig = wasm_bindgen_futures::JsFuture::from(js_sys::Promise::from(sig));
-        let sig = sig.await.map_err(|_| "Bad async result")?;
+        let sig = sig.await.map_err(|_| Error::msg("JavaScript signer throws an error asynchronously."))?;
```

Comment:
Same here, more error details would be better.

Reply:
Done.